### ISSUE TITLE
Error generating archive with archive_file when symlink is present and exclude_symlink_directories is set to true

### ISSUE THREAD

**Issue by sm0ke21** [hashicorp/terraform-provider-archive#296]
### Terraform CLI and Provider Versions

Terraform v1.6.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/archive v2.4.1
+ provider registry.terraform.io/hashicorp/aws v5.32.1
+ provider registry.terraform.io/hashicorp/random v3.6.0

### Terraform Configuration

```terraform
data "archive_file" "admin_zip" {
  type                        = "zip"
  source_dir                  = "${path.module}/../lambda/admin_api/src"
  output_file_mode            = "0666"
  exclude_symlink_directories = true
  output_path                 = "${path.module}/../lambda/admin_api/out/admin.zip"
}
```

### Expected Behavior

I expect a zip file in `output_path` containing everything in `source_dir`

### Actual Behavior

I get the following error:

```
│ Error: Archive creation error
│ 
│   with data.archive_file.admin_zip,
│   on lambda.tf line 20, in data "archive_file" "admin_zip":
│   20: data "archive_file" "admin_zip" {
│ 
│ error creating archive: error archiving directory: error reading file for archival: read ../lambda/admin_api/src/commonlib: is a directory
```

### Steps to Reproduce

1. `terraform plan`

### How much impact is this issue causing?

High

### Logs

https://gist.github.com/sm0ke21/c1371dede8e6e288da456854582d7a00

### Additional Information

There is a symlink inside admin_api/src called `commonlib` linking to common python libraries.
If the symlink is removed then `terraform plan` works.

I expected this to not fail since `exclude_symlink_directories` is set to true.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sm0ke21** [comment]
`commonlib` is a valid symlink

**Comment by bendbennett** [comment]
@sm0ke21 this bug should be fixed in `v2.4.2`:

- https://github.com/hashicorp/terraform-provider-archive/releases/tag/v2.4.2
- https://registry.terraform.io/providers/hashicorp/archive/latest
